### PR TITLE
Fix Telegram API result types

### PR DESCRIPTION
**Pull request by johnspade**

### CODE REVIEW

**Comment by sherzodv** [comment]
@johnspade can you please look through this? I'll merge on your approval

**Comment by sherzodv** [comment]
BTW, I think it's better to have Option[Result] instead of Either[Boolean, Result]. Because, as we discussed Boolean in Tg API here is effectively single valued.

File: iozhik/src/main/scala/io/github/iozhik/ScalaApiGeneratorV1.scala [hunk 1]
```diff
@@ -909,6 +931,14 @@ class ScalaApiGeneratorV1 extends Generator {
           |class ${name}Http4sImp[F[_]: ConcurrentEffect: ContextShift](http: Client[F], baseUrl: String, blocker: Blocker)
           |                          (implicit F: MonadError[F, Throwable]) extends $name[F] {
           |
+          |  implicit def _decodeEither[A, B](implicit
+          |                                   a: io.circe.Decoder[A],
+          |                                   b: io.circe.Decoder[B]): io.circe.Decoder[Either[A, B]] = {
+          |    val l: io.circe.Decoder[Either[A, B]] = a.map (Left.apply)
+          |    val r: io.circe.Decoder[Either[A, B]] = b.map (Right.apply)
+          |    l or r
+          |  }
```
Comment (johnspade) [inline iozhik/src/main/scala/io/github/iozhik/ScalaApiGeneratorV1.scala:966]: The decoder can be made more concise:
```
implicit def decodeEither[A, B](implicit
    decoderA: Decoder[A],
    decoderB: Decoder[B]
  ): Decoder[Either[A, B]] = decoderA.either(decoderB)
```

**Review by johnspade — COMMENTED** [review]
ApiHttp4sImp still tries to decode whole JSON, we need to use the 'result' field

**Comment by johnspade** [comment]
> BTW, I think it's better to have Option[Result] instead of Either[Boolean, Result]. Because, as we discussed Boolean in Tg API here is effectively single valued.

Well, yes, but I still think that Either[Boolean, Result] is more obvious than `None == (Boolean = true)`.

File: iozhik/src/main/scala/io/github/iozhik/ScalaApiGeneratorV1.scala (hunk 1 above)
Comment (sherzodv) [reply iozhik/src/main/scala/io/github/iozhik/ScalaApiGeneratorV1.scala:966]: Nice, we'll go with this one.

**Comment by sherzodv** [comment]
> ApiHttp4sImp still tries to decode whole JSON, we need to use the 'result' field

Yeah, you caught a tough one here. We may need some syntax to handle the case, smth like:

```
def callMe[A]... => {
  result: A (expose)
  ok: Boolean (error indicator)
  description: Option[String] (error description)
}
```

But, these requires long run, so I've just hardcoded this into generator for now. More checks are welcome.

**Comment by johnspade** [comment]
It works.

LGTM (I think we can merge this)

**Pull request merged**